gsa-website/src/components/CardData.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import {Box,Card,CardContent,Typography,Button} from '@mui/material'
+import { getFirestore, collection, doc, setDoc, onSnapshot } from "firebase/firestore";
+import { useState } from "react";
+import { auth } from '../firebase';
+import Authenticate from './Authenticate';
+//Temperature
+import AcUnitTwoToneIcon from '@mui/icons-material/AcUnitTwoTone'; //Low Temp - snowflake 
+import WbSunnyTwoToneIcon from '@mui/icons-material/WbSunnyTwoTone'; // Good Temp - sun 
+import LocalFireDepartmentTwoToneIcon from '@mui/icons-material/LocalFireDepartmentTwoTone'; // Too Hot Temp, fire 
+//Moisture
+import WaterDropTwoToneIcon from '@mui/icons-material/WaterDropTwoTone'; // Low Moisture - just one drop 
+import WavesTwoToneIcon from '@mui/icons-material/WavesTwoTone'; // Good Moisture - water 
+//Sunlight
+import BedtimeTwoToneIcon from '@mui/icons-material/BedtimeTwoTone'; // Low Sunglight - the moon (cresent) 
+import EmojiObjectsTwoToneIcon from '@mui/icons-material/EmojiObjectsTwoTone'; // Good Sunlight - lightbulb
+import LightModeTwoToneIcon from '@mui/icons-material/LightModeTwoTone'; // Too much  Sunlight - the sun 
+
+
+
+
+
+const CardData = () => {
+    const [hubTime, setHubTime] = useState('');
+    const [hubMoisture, setHubMoisture] = useState('');
+    const [hubSunlight, setHubSunlight] = useState('');
+    const [hubTemperature, setHubTemperature] = useState('');
+
+
+    const firestore = getFirestore();
+
+
+    const unsub = onSnapshot(doc(firestore, "HUBS_ONLINE", "HUB_1"), (doc) => {
```

Comment:
https://stackoverflow.com/questions/59944658/which-react-hook-to-use-with-firestore-onsnapshot